Framework/CurveFitting/src/Functions/DecoupAsymPowderMagLong.cpp

```diff
@@ -45,9 +45,11 @@ void DecoupAsymPowderMagLong::functionDeriv1D(Jacobian *out, const double *xValu
   const double charField = getParameter("CharField");
 
   for (size_t i = 0; i < nData; i++) {
-    auto A_z = getAz(xValues[i], charField);
-    double diffasym = A_z;
-    double diffAz = asym;
+    double diffasym = getAz(xValues[i], charField);
+    double diffcharfield = -xValues[i] / pow(charField, 2);
+    double b = xValues[i] / charField;
+    double diffb = ((pow(b, 2) - 1) * ((pow(b, 2) + 3) * log((b + 1.0) / (b - 1.0)) - (2 * b))) / (8 * pow(b, 4));
```

Comment:
this could go in the helper too

Reply:
along with diffcharfield

Reply:
Added a new getDiffAz method to the helper.